bitnami/redis/templates/headless-svc.yaml

```
@@ -7,8 +7,15 @@ metadata:
     {{- if .Values.commonLabels }}
     {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
     {{- end }}
+  annotations:
   {{- if .Values.commonAnnotations }}
-  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+  {{- if .Values.useExternalDNS.enabled }}
+    {{ .Values.useExternalDNS.annotationKey }}hostname: {{ include "redis.externalDNS.suffix" . }}
+    {{- range $key, $val := .Values.useExternalDNS.additionalAnnotations }}
+    {{ $.Values.useExternalDNS.annotationKey }}{{ $key }}: {{ $val | quote }}
+    {{- end }}
```

```suggestion
    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
  {{- end }}
  {{- if .Values.useExternalDNS.enabled }}
    {{ .Values.useExternalDNS.annotationKey }}hostname: {{ include "redis.externalDNS.suffix" . }}
    {{- range $key, $val := .Values.useExternalDNS.additionalAnnotations }}
    {{ $.Values.useExternalDNS.annotationKey }}{{ $key }}: {{ $val | quote }}
    {{- end }}
```

We can notch the indentation of the `commonAnnoations` render. Ideally, we could do the same for the `additionalAnnotations` , however, the spaces currently matter within the `range` loop because of how this is composed within.

What do you think of making this into a single template for all of these annotations, say `redis.externalDNS.annotations`, and instead simply gating this here?

---

@WarheadsSE thanks for the suggestion, I proceeded to do exactly this in ba304f9c5